src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/GestureManager.shared.cs

```diff
@@ -0,0 +1,565 @@
+				: hoverState == HoverState.Hovered
+					? nameof(HoverState.Hovered)
+					: nameof(TouchState.Normal);
+
+			VisualStateManager.GoToState(visualElement, state);
+		}
+
+		async Task SetBackgroundImageAsync(TouchEffect sender, TouchState touchState, HoverState hoverState, int duration, CancellationToken token)
+		{
+			var normalBackgroundImageSource = sender.NormalBackgroundImageSource;
+			var pressedBackgroundImageSource = sender.PressedBackgroundImageSource;
+			var hoveredBackgroundImageSource = sender.HoveredBackgroundImageSource;
+
+			if (normalBackgroundImageSource == null &&
+				pressedBackgroundImageSource == null &&
+				hoveredBackgroundImageSource == null)
+				return;
+
+			var aspect = sender.BackgroundImageAspect;
+			var source = normalBackgroundImageSource;
+			if (touchState == TouchState.Pressed)
+			{
+				if (sender.Element.IsSet(TouchEffect.PressedBackgroundImageAspectProperty))
+					aspect = sender.PressedBackgroundImageAspect;
+
+				source = pressedBackgroundImageSource;
+			}
+			else if (hoverState == HoverState.Hovered)
+			{
+				if (sender.Element.IsSet(TouchEffect.HoveredBackgroundImageAspectProperty))
+					aspect = sender.HoveredBackgroundImageAspect;
+
+				if (sender.Element.IsSet(TouchEffect.HoveredBackgroundImageSourceProperty))
+					source = hoveredBackgroundImageSource;
+			}
+			else
+			{
+				if (sender.Element.IsSet(TouchEffect.NormalBackgroundImageAspectProperty))
+					aspect = sender.NormalBackgroundImageAspect;
+			}
+
+			try
+			{
+				if (sender.ShouldSetImageOnAnimationEnd)
+					await Task.Delay(duration, token);
+			}
+			catch (TaskCanceledException)
+			{
+				return;
+			}
+
+			if (sender.Element is Image image)
+			{
+				using (image.Batch())
+				{
+					image.Aspect = aspect;
+					image.Source = source;
+				}
+			}
+		}
+
+		Task SetBackgroundColor(TouchEffect sender, TouchState touchState, HoverState hoverState, int duration, Easing easing)
+		{
+			var normalBackgroundColor = sender.NormalBackgroundColor;
+			var pressedBackgroundColor = sender.PressedBackgroundColor;
+			var hoveredBackgroundColor = sender.HoveredBackgroundColor;
+
+			if (normalBackgroundColor == Color.Default &&
+				pressedBackgroundColor == Color.Default &&
+				hoveredBackgroundColor == Color.Default)
+				return Task.FromResult(false);
+
+			var control = sender.Element;
+			if (defaultBackgroundColor == default)
+				defaultBackgroundColor = control.BackgroundColor;
+
+			var color = GetBackgroundColor(normalBackgroundColor);
+
+			if (touchState == TouchState.Pressed)
+				color = GetBackgroundColor(pressedBackgroundColor);
+			else if (hoverState == HoverState.Hovered && sender.Element.IsSet(TouchEffect.HoveredBackgroundColorProperty))
+				color = GetBackgroundColor(hoveredBackgroundColor);
+
+			if (duration <= 0)
+			{
+				control.BackgroundColor = color;
+				return Task.FromResult(true);
+			}
+
+			var animationCompletionSource = new TaskCompletionSource<bool>();
+			new Animation
```

Comment:
This is very nice, but, what if we create some extensions to animate certain properties like colors, etc? (Example: https://github.com/jsuarezruiz/Xamanimation/blob/master/Xamanimation/Extensions/ColorExtensions.cs)
It will be a little bit more effort but, we can have extensions to animate certain properties in a simple way.

Reply:
Done